### PR TITLE
[DOCS] Reformat token count limit filter docs

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the token count limit token filter docs as part of #44726.

### Changes

* Adds a title abbreviation
* Updates the description and adds a Lucene link
* Add tip directing users to `ignore_above` mapping parameter
* Reformats the parameters section
* Adds analyze, custom analyzer, and custom filter snippets

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc [hunk 1]
```diff
@@ -1,32 +1,134 @@
...
-is `false`.
-|=======================================================================
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/miscellaneous/LimitTokenCountFilter.html[LimitTokenCountFilter].
 
-Here is an example:
+[TIP]
+====
+The `limit` filter is commonly used to limit the size of document field values
+based on _token count_. If you want to limit the size of field values based on
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc]: The first sentence in this tip seems like it belongs in the description as the "why" you want to use this filter.

File: docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc [hunk 2]
```diff
@@ -1,32 +1,134 @@
...
+          "filter": [ "limit" ]
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+[[analysis-limit-token-count-tokenfilter-configure-parms]]
+==== Configurable parameters
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc]: Should this come before the Example section? (And maybe include the "basic" example up front?)

**Review by debadair — APPROVED** [review]
Incorporate or ignore as you see fit. :-)

File: docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc]: Thanks! Moved this with 64239c99fe9.

File: docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/limit-token-count-tokenfilter.asciidoc]: Relocated with c353875b6b2.

**Comment by jrodewig** [comment]
Thanks @debadair!

**Comment by jrodewig** [comment]
master: 9907b0a
7.x: cd04021961515309e1cc47018f147899cabd80a4
7.5: 8a200b002e436f250b6811c5628045992eb8d3ad

**Pull request merged**